sdk/core/azure-core/azure/core/pipeline/policies/_retry.py

```diff
@@ -263,12 +263,20 @@ def is_retry(self, settings, response):
         code is on the list of status codes to be retried upon on the
         presence of the aforementioned header.
 
+        The behavior is:
+        -	If status_code < 400: don’t retry
+        -	Else if Retry-After present: retry
+        -	Else: retry based on the safe status code list ([408, 429, 500, 502, 503, 504])
+
+
         :param dict settings: The retry settings.
         :param response: The PipelineResponse object
         :type response: ~azure.core.pipeline.PipelineResponse
         :return: True if method/status code is retryable. False if not retryable.
         :rtype: bool
         """
+        if response.http_response.status_code and response.http_response.status_code < 400:
```

Comment:
If status_code is not an integer, we have bigger problem :). I would just do 
```python
if response.http_response.status_code < 400:
```

Reply:
I add this because some of our tests do not specify response.http_response.status_code. So w/o the check, tests will be broken.